packages/playground/src/playground-db.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+defined('ABSPATH') || exit;
+
+function playground_escape_array($array)
+{
+	global $wpdb;
+	$escaped = array();
+	foreach ($array as $value) {
+		if (is_numeric($value))
+			$escaped[] = $wpdb->prepare('%d', $value);
+		else
+			$escaped[] = $wpdb->prepare('%s', $value);
+	}
+	return implode(',', $escaped);
+}
+
+function playground_dump_db($zip)
+{
+	global $wpdb;
+
+	$tables   = playground_get_db_tables();
+	$sql_dump = array();
+
+	foreach ($tables as $table) {
+		array_push(
+			$sql_dump,
+			sprintf("DROP TABLE IF EXISTS `%s`;", esc_sql($table)),
+			playground_dump_db_schema($table)
+		);
+	}
+
+	foreach ($tables as $table) {
+		$records = $wpdb->get_results(
+			sprintf(
+				'SELECT * FROM `%s`',
+				esc_sql($table)
+			),
+			ARRAY_A
+		);
+
+		if ($wpdb->last_error) {
+			error_log($wpdb->last_error);
+			continue;
+		}
+
+		foreach ($records as $record) {
+			array_push(
+				$sql_dump,
+				sprintf(
+					'INSERT INTO `%1$s` (%2$s) VALUES (%3$s);',
+					esc_sql($table),
+					playground_escape_array(
+						array_keys($record)
+					),
+					playground_escape_array(array_values($record))
+				)
+			);
+		}
+	}
+	$zip->addFile('schema/_Schema.sql', implode("\n", $sql_dump));
+
+}
+
+function playground_get_db_tables()
+{
+	global $wpdb;
+	$tables = $wpdb->get_results('SHOW TABLES', ARRAY_N);
+	return array_map(fn ($t) => $t[0], $tables);
```

Comment:
Arrow functions were introduced in PHP 7.4, let's make sure this plugin runs on the same PHP versions as WordPress core, that is >= 7.0. How easy would it be to add a phpcs check for that?

Reply:
Done in  e0cb7c58fe6785d24fbc53006677c1d8f5667f76